src/main/java/io/quarkus/gizmo/MethodCheckerVisitor.java

```diff
@@ -0,0 +1,220 @@
+package io.quarkus.gizmo;
+
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.Attribute;
+import org.objectweb.asm.Handle;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.TypePath;
+
+public class MethodCheckerVisitor extends MethodVisitor {
+    private boolean isReturningValue;
+    private boolean isSuperOrThisCalled;
+
+    public MethodCheckerVisitor() {
+        this(Gizmo.ASM_API_VERSION);
+    }
+    public MethodCheckerVisitor(int api) {
+        super(api);
+        isReturningValue = false;
+        isSuperOrThisCalled = false;
+    }
+
+
+    public void check(MethodCreatorImpl methodCreator) {
+        visitLabel(methodCreator.getTop());
+        for (BytecodeCreatorImpl.Operation op : methodCreator.operations) {
+            op.writeBytecode(this);
+        }
+        visitLabel(methodCreator.getBottom());
+    }
+
+    @Override
+    public void visitInsn(int opcode) {
+        // if returnValue or ThrowException it is valid method.
+        switch (opcode) {
+            case Opcodes.RETURN:
+            case Opcodes.IRETURN:
+            case Opcodes.LRETURN:
+            case Opcodes.FRETURN:
+            case Opcodes.DRETURN:
+            case Opcodes.ARETURN:
+            case Opcodes.ATHROW:
+                isReturningValue = true;
+                break;
+        }
+    }
+
+    @Override
+    public void visitParameter(String name, int access) {
+    }
+
+    @Override
+    public AnnotationVisitor visitAnnotationDefault() {
+        return new DummyVisitor();
+    }
+
+    @Override
+    public AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
+        return new DummyVisitor();
+    }
+
+    @Override
+    public AnnotationVisitor visitTypeAnnotation(int typeRef, TypePath typePath, String descriptor, boolean visible) {
+        return new DummyVisitor();
+    }
+
+    @Override
+    public void visitAnnotableParameterCount(int parameterCount, boolean visible) {
+    }
+
+    @Override
+    public AnnotationVisitor visitParameterAnnotation(int parameter, String descriptor, boolean visible) {
+        return new DummyVisitor();
+    }
+
+    @Override
+    public void visitAttribute(Attribute attribute) {
+    }
+
+    @Override
+    public void visitCode() {
+    }
+
+    @Override
+    public void visitFrame(int type, int numLocal, Object[] local, int numStack, Object[] stack) {
+    }
+
+    @Override
+    public void visitIntInsn(int opcode, int operand) {
+    }
+
+    @Override
+    public void visitVarInsn(int opcode, int var) {
+    }
+
+    @Override
+    public void visitTypeInsn(int opcode, String type) {
+    }
+
+    @Override
+    public void visitFieldInsn(int opcode, String owner, String name, String descriptor) {
+    }
+
+    @Override
+    public void visitMethodInsn(int opcode, String owner, String name, String descriptor) {
+    }
```

Comment:
you don't need isSuperOrThisCalled = true; like other visitMethodInsn ?

Reply:
ok I just added it.